### PR TITLE
Prevent duplicate pending invitations per workspace

### PR DESCRIPTION
**Pull request by geclos**
## Summary
- add a guard in `createInviteIntentUseCase` to check existing pending invites for the same workspace
- return a typed 409 `InviteAlreadyPendingError` when the same email already has a pending invite in that workspace
- add unit tests covering normal invite creation, duplicate-in-workspace rejection, and allow-in-different-workspace behavior

## Testing
- `pnpm --filter @domain/auth exec vitest run src/use-cases/create-invite-intent.test.ts`
- `pnpm --filter @domain/auth check`
- `pnpm --filter @domain/auth typecheck`

<div><a href="https://cursor.com/agents/bc-21edf118-1003-46e2-8167-b1dbd6bfe71c"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-21edf118-1003-46e2-8167-b1dbd6bfe71c"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

### CODE REVIEW

File: packages/domain/auth/src/use-cases/create-invite-intent.ts [hunk 1]
```diff
@@ -1,9 +1,18 @@
+import { OrganizationId } from "@domain/shared"
 import { UserRepository } from "@domain/users"
-import { Effect } from "effect"
+import { Data, Effect } from "effect"
 import { createAuthIntent } from "../entities/auth-intent.ts"
 import { AuthIntentRepository } from "../ports/auth-intent-repository.ts"
 import { createInviteIntentData, normalizeEmail } from "./auth-intent-policy.ts"
 
+export class InviteAlreadyPendingError extends Data.TaggedError("InviteAlreadyPendingError")<{
```
Comment (andresgutgon) [inline packages/domain/auth/src/use-cases/create-invite-intent.ts:8]: Didn't we have a helper for doing errors now?

Comment (geclos) [reply packages/domain/auth/src/use-cases/create-invite-intent.ts:8]: nein, the code was too hacky, i've added a agents.md rule and fixed all offending instances for now
